packages/core/src/values/ValueDB.ts

```diff
@@ -160,11 +160,26 @@ export class ValueDB extends EventEmitter {
 		this.nodeId = nodeId;
 		this._db = valueDB;
 		this._metadata = metadataDB;
+
+		this._index = this.buildIndex();
 	}
 
 	private nodeId: number;
 	private _db: JsonlDB<unknown>;
 	private _metadata: JsonlDB<ValueMetadata>;
+	private _index: Set<string>;
+
+	private buildIndex(): Set<string> {
+		const ret = new Set<string>();
+		for (const key of this._db.keys()) {
+			if (compareDBKeyFast(key, this.nodeId)) ret.add(key);
+		}
+		for (const key of this._metadata.keys()) {
+			if (!ret.has(key) && compareDBKeyFast(key, this.nodeId))
+				ret.add(key);
+		}
```

Comment:
Use normal for loops with index, is much faster then for-of

Reply:
Is the index build only once at startup ? In that case it is much more preferred to have fast lookups at runtime and accept the few seconds at startup.

Reply:
Not possible. `Map.keys()` returns an iterator.

Reply:
Great job btw so far hunting this down!

Reply:
Somehow I missed your comment @marcelveldt. Yes the index is built only once at startup but it currently does (# of nodes)x the work that is necessary.
So I'm going to look into that as a last step.